plugins/org.jboss.tools.openshift.egit.core/src/org/jboss/tools/openshift/egit/core/EGitUtils.java

```diff
@@ -229,7 +229,11 @@ public static Repository createRepository(IProject project,
 	}
 
 	public static File getRepositoryPathFor(IProject project) {
-		return new File(project.getLocationURI().getPath(), Constants.DOT_GIT);
+		File f = new File(project.getLocationURI().getPath());
```

Comment:
It's not a matter of complexity but a matter of consistency. If EGit does a search for the repo the way it does then there are valid reasons for it. I dont see why we should do an alternative approach as EGit is the authoritative opinion here tbh.